sdk/src/nonce_state.rs

```diff
@@ -129,6 +127,34 @@ impl<'a> NonceAccount for KeyedAccount<'a> {
 
         Ok(())
     }
+
+    fn initialize(
+        &mut self,
+        recent_blockhashes: &RecentBlockhashes,
+        rent: &Rent,
+        signers: &HashSet<Pubkey>,
+    ) -> Result<(), InstructionError> {
+        if recent_blockhashes.is_empty() {
+            return Err(NonceError::NoRecentBlockhashes.into());
+        }
+
+        if !signers.contains(self.unsigned_key()) {
```

Comment:
why does initialize require an authority?

where is the authorized noncer set?

Reply:
> why does initialize require an authority?

It seems like any modification of account state should occur under _some_ authority.  What would be gained by allowing third-party initialization?

> where is the authorized noncer set?

Next PR.  This one was just ground work

Reply:
> > why does initialize require an authority?
> 
> It seems like any modification of account state should occur under _some_ authority. What would be gained by allowing third-party initialization?

it allows for derived account addresses, e.g. if create_account() allowed a signer to create accounts in some space of addresses derived from the 'from' pubkey

Reply:
Ah!  I didn't realize these were going to share an interface.  We can still verify that `DeriveAddress(authority_pubkey, index, ...) == self.unsigned_pubkey()`, no?

Reply:
I'm saying that it's unnecessary to enforce, and adds complexity for cases like where we have derived account addresses

Reply:
the move from create_account() -> nonce_init() is a single TX, there's no exposure...

also: an uninitialized nonce can't do anything but move from uninit to init, so why ask who's moving it?

Reply:
Yeah, I see where you're coming from.  I suspect it leaves us open to namespace griefing, but I'm not sure that's a problem we need to worry about at this point.  Removed in https://github.com/solana-labs/solana/pull/7378/commits/749f803b8c049b59685e9004532a6e0047a9d5e8